src/parthenon_manager.cpp

```diff
@@ -112,6 +116,26 @@ ParthenonStatus ParthenonManager::ParthenonInitEnv(int argc, char *argv[]) {
   // Set the global number of ghost zones
   Globals::nghost = pinput->GetOrAddInteger("parthenon/mesh", "nghost", 2);
 
+  // set sparse config
+  Globals::sparse_config.enabled = pinput->GetOrAddBoolean(
+      "parthenon/sparse", "enable_sparse", Globals::sparse_config.enabled);
+#ifndef ENABLE_SPARSE
+  PARTHENON_REQUIRE_THROWS(
+      !Globals::sparse_config.enabled,
+      "Sparse is compile-time disabled but was requested to be enabled in input file");
+#endif
+  Globals::sparse_config.allocation_threshold = pinput->GetOrAddReal(
+      "parthenon/sparse", "alloc_threshold", Globals::sparse_config.allocation_threshold);
+  Globals::sparse_config.deallocation_threshold =
+      pinput->GetOrAddReal("parthenon/sparse", "dealloc_threshold",
+                           Globals::sparse_config.deallocation_threshold);
+  Globals::sparse_config.deallocation_count = pinput->GetOrAddInteger(
+      "parthenon/sparse", "dealloc_count", Globals::sparse_config.deallocation_count);
```

Comment:
NONBLOCKING: As discussed before. If we go to per variable allocation thresholds, this will need to change.